src/providers/bibtexformater.ts

```diff
@@ -35,7 +61,20 @@ export class BibtexFormater {
             right: leftright[1],
             sort: config.get('bibtex-format.sortby') as string[]
         }
+        const lineOffset = range ? range.start.line : 0
+        const columnOffset = range ? range.start.character : 0
 
+        const ast = await this.extension.pegParser.parseBibtex(document.getText(range)).catch((error) => {
+            if (error instanceof(Error)) {
+                this.extension.logger.addLogMessage('Bibtex parser failed.')
+                this.extension.logger.addLogMessage(error.message)
+                this.extension.logger.showErrorMessage('Bibtex parser failed with error: ' + error.message)
+            }
+            return undefined
+        })
+        if (! ast) {
+            return []
```

Comment:
With the error handling, when calling `latex-workshop.bibalign`, the message `BibTeX action successful. ` will be logged even if it fails.
```
[21:47:22] Start bibtex formatting on user request.
[21:47:22] Bibtex parser failed.
[21:47:22] Expected "=", "\r\n", or [ \t\n] but "a" found.
[21:47:22] BibTeX action successful. Took 33.06112098693848 ms.
```

Reply:
Thanks for all this. This is a kind of a headache to allow both a direct usage and the formatter approach.